functional_consumergroup_test.go

```diff
@@ -270,6 +277,80 @@ func TestConsumergroupWatchInstances(t *testing.T) {
 	}
 }
 
+func TestConsumergroupInstanceWatchRegistration(t *testing.T) {
+	kz, err := NewKazoo(zookeeperPeers, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer assertSuccessfulClose(t, kz)
+
+	cg := kz.Consumergroup("test.kazoo.TestConsumergroupWatchRegistration")
+	if err := cg.Create(); err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		if err := cg.Delete(); err != nil {
+			t.Error(err)
+		}
+	}()
+
+	instance := cg.NewInstance()
+	if err := instance.Register([]string{"topic"}); err != nil {
+		t.Fatal(err)
+	}
+
+	// Set a watch and make sure that it does not trigger on its own.
+	reg, c, err := instance.WatchRegistration()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if reflect.DeepEqual(reg.Subscription, map[string]int{}) {
+		t.Fatalf("Bad subscription, got=%v", reg.Subscription)
+	}
+	select {
+	case <-c:
+		t.Fatal("Watch is not supposed to be triggered")
+	case <-time.After(200 * time.Millisecond):
+	}
+
+	// The watch is triggered by an update.
+	if err = instance.Register([]string{"foo", "bar"}); err != nil {
+		t.Fatal(err)
+	}
+	<-c
```

Comment:
My only nitpick is that if the channel is never closed this will block forever without informing that the test failed.

Reply:
I will add a timeout for that.

Reply:
Fixed